bioframe/core.py

```diff
@@ -0,0 +1,684 @@
+    - columns have valid dtypes (object/string/categorical, int, int)
+    - all starts < ends.
+
+    raise_errors:bool
+        If true, raises errors instead of returning a boolean False for invalid properties.
+        Default false.
+
+    cols : (str, str, str) or None
+        The names of columns containing the chromosome, start and end of the
+        genomic intervals, provided separately for each set. The default
+        values are 'chrom', 'start', 'end'.
+
+    Returns
+    -------
+    is_bedframe:bool
+
+    """
+    ck1, sk1, ek1 = _get_default_colnames() if cols is None else cols
+
+    if not _verify_columns(df, [ck1, sk1, ek1], return_as_bool=True):
+        if raise_errors:
+            raise TypeError("Invalid columns")
+        return False
+
+    if not _verify_column_dtypes(df, cols=[ck1, sk1, ek1], return_as_bool=True):
+        if raise_errors:
+            raise TypeError("Invalid column dtypes")
+        return False
+
+    if ((df[ek1] - df[sk1]) < 0).any():
+        if raise_errors:
+            raise ValueError(
+                "Invalid genomic interval dataframe: starts exceed ends for "
+                + str(np.sum(((df[ek1] - df[sk1]) < 0)))
+                + " intervals"
+            )
+        return False
+
+    return True
+
+
+def is_overlapping(df, cols=None):
+    """
+    tests if any genomic intervals in a bioframe `df` overlap
+
+    Returns
+    -------
+    is_overlapping:bool
+
+    """
+
+    ck1, sk1, ek1 = _get_default_colnames() if cols is None else cols
+
+    df_merged = ops.merge(df, cols=cols)
+
+    total_interval_len = np.sum((df[ek1] - df[sk1]).values)
+    total_interval_len_merged = np.sum((df_merged[ek1] - df_merged[sk1]).values)
+
+    if total_interval_len > total_interval_len_merged:
+        return True
+    else:
+        return False
+
+
+def is_viewframe(region_df, raise_errors=False, view_name_col="name", cols=None):
+    """
+    Checks that region_df is a valid view, namely that it:
+    - satisfies requirements for a bedframe, including columns for ('chrom', 'start', 'end')
+    - has an additional column providing a unique name, view_name_col, with default 'name'
+    - does not contain null values
+    - that entries in the view_name_col are unique.
+    - intervals are non-overlapping
+
+    raise_errors:bool
+        If true, raises errors instead of returning a boolean for invalid properties.
+        Default false.
+
+    Returns
+    -------
+    is_viewframe:bool
+
+    """
+
+    ck1, sk1, ek1 = _get_default_colnames() if cols is None else cols
+
+    if not _verify_columns(
+        region_df, [ck1, sk1, ek1, view_name_col], return_as_bool=True
+    ):
+        if raise_errors:
+            raise TypeError("Invalid view: column names cannot be verified")
+        return False
+
+    if not is_bedframe(region_df, cols=cols):
+        if raise_errors:
+            raise ValueError("Invalid view: bedframe properties cannot be verified")
```

Comment:
Could chain exceptions here

Reply:
do you have an example where this is done?